### PR TITLE
Bump librdkafka to 1.9.0-RC1

### PR DESCRIPTION
**Pull request by leonmaia**
There's a fix in the latest version of librdkafka which fixes `rd_kafka_list_groups` hangs.
We experience this issue on several occasions in our consumers, so it would be very important to get this fix applied.

For more context, here's the [fix PR](https://github.com/edenhill/librdkafka/issues/3705#issuecomment-1028728359)

Given that the fix is only now present in RC-releases of the version 1.9.0, can we release a `beta.1` version of rdkafka-ruby containing the fix?

### CODE REVIEW

**Comment by thijsc** [comment]
Yeah we could! Did you take a look at the failing tests, are they caused by the upgrade?

**Comment by leonmaia** [comment]
@thijsc it seems that there's an integration issue with the current `1.9.0` and the changes made by #172 .

**Comment by jvortmann** [comment]
@leonmaia could you identify was were the integration issue?
How can I reproduce it locally to debug?

**Comment by leonmaia** [comment]
I'm able to get quite consistent errors when running `bundle exec rspec spec/rdkafka/producer/client_spec.rb` locally @jvortmann .

TBH I'm not even sure if this is also related to the upgrade to `1.9.0` upgrade or simply flakiness in the test suite. If I have time I want to test doing some random change and just triggering the tests again.

The issue really seems to be this spec `spec/rdkafka/producer/client_spec.rb` since with the revert I was able to get green builds every time.

**Comment by leonmaia** [comment]
Hi @thijsc, I also added the flakiness fix to this PR in case you think it's enough to merge just this one.

File: spec/rdkafka/producer/client_spec.rb [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+  let(:native_config) { config.send(:native_config) }
+  let(:native_kafka)  { config.send(:native_kafka, native_config, :rd_kafka_consumer) }
 
-  subject(:client) { described_class.new(native) }
+  subject(:client) { described_class.new(native_kafka) }
 
   before do
-    allow(Rdkafka::Bindings).to receive(:rd_kafka_poll).with(native, 250)
-    allow(Rdkafka::Bindings).to receive(:rd_kafka_outq_len).with(native).and_return(0)
+    puts "#{native_kafka}"
```
Comment (jvortmann) [inline spec/rdkafka/producer/client_spec.rb:13]: There is a missing `puts` on the setup.

File: spec/rdkafka/producer/client_spec.rb [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+  let(:closing)       { false }
+  let(:config)        { rdkafka_consumer_config }
+  let(:native_config) { config.send(:native_config) }
+  let(:native_kafka)  { config.send(:native_kafka, native_config, :rd_kafka_consumer) }
 
-  subject(:client) { described_class.new(native) }
+  subject(:client) { described_class.new(native_kafka) }
 
   before do
-    allow(Rdkafka::Bindings).to receive(:rd_kafka_poll).with(native, 250)
```
[the remark this answers is not in the document]
Comment (jvortmann) [reply spec/rdkafka/producer/client_spec.rb:11]: https://appsignal.semaphoreci.com/jobs/c27149b7-8170-4f98-a958-578f35964e92

**Review by jvortmann — COMMENTED** [review]
Added some notes about the spec failure.

**Comment by leonmaia** [comment]
Thanks @jvortmann , it seems with the changes I made the errors are less frequent but they are still reproduceable.

**Comment by jvortmann** [comment]
@leonmaia created a PR to fix the issue on top of master on #189 . Can you rebase your changes on top of that and check, please?

**Comment by jvortmann** [comment]
> Thanks @jvortmann , it seems with the changes I made the errors are less frequent but they are still reproduceable.

I could reproduce them locally (the same error you saw on the CI) and fix those.
I eventually got another one were the expectation is over 1 call on `rd_kafka_poll` but got 3 calls.

**Comment by leonmaia** [comment]
@thijsc I rebased with the latest `main`.

**Pull request merged**